src/main/java/io/lumigo/core/SpansContainer.java

```diff
@@ -354,14 +360,88 @@ public void addHttpSpan(Long startTime, Request<?> request, Response<?> response
+                                                                                        .headers())))
+                                                .uri(
+                                                        callIfVerbose(
+                                                                () ->
+                                                                        context.httpRequest()
+                                                                                .getUri()
+                                                                                .toString()))
+                                                .method(context.httpRequest().method().name())
+                                                .body(
+                                                        callIfVerbose(
+                                                                () ->
+                                                                        extractBodyFromRequest(
+                                                                                context
+                                                                                        .requestBody())))
+                                                .build())
+                                .response(
+                                        HttpSpan.HttpData.builder()
+                                                .headers(
+                                                        callIfVerbose(
+                                                                () ->
+                                                                        extractHeadersV2(
+                                                                                context.httpResponse()
+                                                                                        .headers())))
+                                                .body(
+                                                        callIfVerbose(
+                                                                () ->
+                                                                        extractBodyFromResponse(
+                                                                                context
+                                                                                        .response())))
+                                                .statusCode(context.httpResponse().statusCode())
+                                                .build())
+                                .build());
+
+        Logger.debug(
+                "Trying to extract aws custom properties for service: "
+                        + executionAttributes.getAttribute(SdkExecutionAttribute.SERVICE_NAME));
+        AwsSdkV2ParserFactory.getParser(
+                        executionAttributes.getAttribute(SdkExecutionAttribute.SERVICE_NAME))
+                .safeParse(httpSpan, context);
```

Comment:
are you sure nothing can be shared with v1 functions? painful to see this code duplication...

Reply:
not with the time limits we have now